pippo-core/src/main/java/ro/pippo/core/route/CorsHandlerV3.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ro.pippo.core.route;
+
+import ro.pippo.core.HttpConstants;
+import ro.pippo.core.PippoRuntimeException;
+import ro.pippo.core.util.StringUtils;
+
+/**
+ * Define how CORS requests are handled.
+ *
+ * <p>
+ * The Cross-Origin Resource Sharing standard works by adding new HTTP headers
+ * that allow servers to describe the set of origins that are permitted to read
+ * that information using a web browser.
+ * </p>
+ *
+ * <p>
+ * For more details see: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
+ * </p>
+ */
+public class CorsHandlerV3 implements RouteHandler<RouteContext> {
+
+    private String allowOrigin;
+
+    private String exposeHeaders;
+
+    private int maxAge = -1;
+
+    private boolean allowCredentials;
+
+    private String allowMethods;
+
+    private String allowHeaders;
+
+    /**
+     * The {@code Access-Control-Allow-Origin} response header indicates whether the
+     * response can be shared with requesting code from the given origin.
+     *
+     * @param allowOrigin
+     *            origin, eg: http://pippo.ro
+     */
+    public CorsHandlerV3(String allowOrigin) {
+        if (StringUtils.isNullOrEmpty(allowOrigin)) {
+            throw new PippoRuntimeException("allowOrigin cannot be blank");
+        }
+        this.allowOrigin = allowOrigin;
+    }
+
+    @Override
+    public void handle(RouteContext context) {
+        context.getResponse().header(HttpConstants.Header.ACCESS_CONTROL_ALLOW_ORIGIN, allowOrigin);
```

Comment:
Maybe here to extract the response object and to use it in next lines:
```java
Response response = context.getResponse();
```

Reply:
Done!